lib/glueby/wallet/tapyrus_core_wallet_adapter.rb

```diff
@@ -0,0 +1,127 @@
+# frozen_string_literal: true
+
+require 'securerandom'
+require 'bigdecimal'
+
+module Glueby
+  class Wallet
+    class TapyrusCoreWalletAdapter < AbstractWalletAdapter
+      WALLET_PREFIX = 'wallet-'.freeze
+      ADDRESS_TYPE = 'legacy'.freeze
+
+      RPC_WALLET_NOT_FOUND_ERROR_CODE = '-18'
+
+      def create_wallet
+        wallet_id = SecureRandom.hex(32)
+        Contract::RPC.client.createwallet(wallet_name(wallet_id))
+        wallet_id
+      end
+
+      # On TapyrusCoreWallet, there are no way to delete real wallet via RPC.
+      # So, here just unload.
+      def delete_wallet(wallet_id)
+        unload_wallet(wallet_id)
+      end
+
+      def load_wallet(wallet_id)
+        Contract::RPC.client.loadwallet(wallet_name(wallet_id))
+      end
+
+      def unload_wallet(wallet_id)
+        Contract::RPC.client.unloadwallet(wallet_name(wallet_id))
+      end
+
+      def wallets
+        Contract::RPC.client.listwallets.map do |wallet_name|
+          match = /#{WALLET_PREFIX}(?<wallet_id>.+)/.match(wallet_name)
+          next unless match
+
+          match[:wallet_id]
+        end.compact
```

Comment:
`select` is better than `map`.

Reply:
I think 'map' is better. Here want to extract the wallet id from the wallet name using regex. `select` returns items that are the same with the input list.

Reply:
Oh, I see.